pkg/v1/tkg/tkgctl/helpers.go

```diff
@@ -184,63 +194,329 @@ func (t *tkgctl) checkIfInputFileIsClusterClassBased(clusterConfigFile string) (
+			err = processYamlObjectAndAddToMap(value[index], clusterAttributePath+"."+strconv.Itoa(index), inputVariablesMap)
+		} else if regExMachineDep.MatchString(clusterAttributePath) || clusterAttributePath == constants.TopologyVariables {
+			// The attribute paths - spec.topology.workers.machineDeployments.[0-9].variables.overrides, spec.topology.variables
+			// has array of data, each array element is of type map, with key(as "name")/value (as "value")
+			// to organize better, get "name" and "value", append "name" value to attribute path and "value" value as value, process them as next level.
+			variables := value
+			for varIndex := range variables {
+				nameValueMap := variables[varIndex].(map[string]interface{})
+				varName := nameValueMap["name"]
+				varValue := nameValueMap["value"]
+				nextLevelName := clusterAttributePath + "." + varName.(string)
+				// The attribute path "spec.topology.variables.trust.*" has value of type array, each array element is of type map, map has keys - "name" and "data",
+				// need to process "spec.topology.variables.trust."+ (value of "name") as attribute path and value of "data" next level value, process it again.
+				if varName == "trust" {
+					trustValArr := varValue.([]interface{})
+					for trustIndex := range trustValArr {
+						trustName := trustValArr[trustIndex].(map[string]interface{})["name"].(string)
+						trustData := trustValArr[trustIndex].(map[string]interface{})["data"]
+						err = processYamlObjectAndAddToMap(trustData, nextLevelName+"."+trustName, inputVariablesMap)
+						if err != nil {
+							return err
 						}
 					}
-					if len(value) > 0 {
-						variablesMap[name] = value
-					}
+					continue // we are done processing "trust" variable, so process next variable
+				} else if varName == "TKR_DATA" { // no need to process TKR_DATA, because there is no mapping as of now
+					continue
+				}
+				err = processYamlObjectAndAddToMap(varValue, nextLevelName, inputVariablesMap)
+				if err != nil {
+					return err
 				}
-				t.TKGConfigReaderWriter().SetMap(variablesMap)
 			}
+			break // all variables are processed in above loop so break it.
+		} else {
+			// process specific index value, this index value could be of any type
+			err = processYamlObjectAndAddToMap(value[index], clusterAttributePath, inputVariablesMap)
+		}
+		if err != nil {
+			return err
+		}
+	}
+	return err
+}
+
+// getProviderNameFromTopologyClassName takes input cluster class spec.topology.class value and validates, returns provides name.
+// The Cluster spec.topology.class attribute value should be valid kubernetes object name : it should be string,
+// should be splittable with "-", after split, it should have at least 3 parts, second part should be infra provider name - [aws, azure, vsphere, docker]
+// eg: tkg-aws-default : infra provider is aws.
+// tkg-unknow-cluster1 : this is not valid name,  "unknow" is not valid infra provider name.
```

Comment:
> tkg-unknow-cluster1 : this is not valid name,  "unknow" is not valid infra provider name.

Is this the right assumption to do? What if user has created custom clusterclass which doesn't follow the naming convention we are using?

Reply:
I don't see any other way to identify the provider type from Cluster class, I got requirements from Nick, Nick please add your comment @tenczar

Reply:
If the validations are now only limited to the default CC we install, then lets be explicit about it.
Even then let's not make assumptions on the cluster name used as a means to determine the infra for which it is targetting. Can we annotate the CC with some more reliable metadata?

Reply:
nit unknow->unknown, but as mentioned in earlier comment, it goes beyond just the clustername pattern.

Reply:
yeah, there is nothing in the ccluster definition that gives us the provider. Do we absolutely needed it? What validations require us knowing the provider?

Reply:
I have created a GitHub issue, to validate the cluster class and identify the provider,  https://github.com/vmware-tanzu/tanzu-framework/issues/2424, and added the same in as TODO in code.
Also created Tech Debt Jira https://jira.eng.vmware.com/browse/TKG-11398, to figure out a better way to validate cluster class inputs.